src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -40,7 +40,14 @@ public StashRequestFacade(StashPluginConfiguration stashPluginConfiguration) {
   }
   
   public void initialize(File projectBaseDir) {
-    this.projectBaseDir = projectBaseDir;
+    if (config.getBaseDir() != null && !"".equals(config.getBaseDir()))
```

Comment:
* Is this the correct logic? We should use the basedir from the configuration if it exists, or?
* `StringUtils.isNotEmpty()`/`StringUtils.isEmpty()`
* Please keep the code style consistent (no extra lines for braces)
* This method could be renamed to closer match its function: `setDefaultBaseDir` (open to suggestions)